pkg/controller/backup_controller.go

```diff
@@ -178,9 +188,14 @@ func (c *backupController) processBackup(key string) error {
 		backup.Status.Expiration = metav1.NewTime(c.clock.Now().Add(backup.Spec.TTL.Duration))
 	}
 
-	var backupLocation *api.BackupStorageLocation
-	// validation
-	if backupLocation, backup.Status.ValidationErrors = c.getLocationAndValidate(backup, c.defaultBackupLocation); len(backup.Status.ValidationErrors) > 0 {
+	backupLocation, statusBackup := c.getLocationAndValidate(backup, c.defaultBackupLocation)
+	backup.Status.ValidationErrors = append(backup.Status.ValidationErrors, statusBackup...)
+
+	// TODO: add VSLs to the actual backup
```

Comment:
I'm thinking we don't need to store the `VolumeSnapshotLocation` objects anywhere within the controller.  `pkg/backup` will handle instantiating a BlockStore based on the appropriate location as necessary.  This should make the signature of this function simpler, and it can probably be called from within `getLocationAndValidate`